IntegrationTests/UI.IntegrationTests/CommandsDialogs/FormCommitTests.cs

```diff
@@ -302,6 +349,15 @@ public void editFileToolStripMenuItem_Click_no_selection_should_not_throw()
             });
         }
 
+        private void RunTwoFormsConsecutively(Action<FormCommit, FormCommit> testDriver)
+        {
+            FormCommit form1 = null;
+            FormCommit form2 = null;
+            RunFormTest(form => form1 = form);
+            RunFormTest(form => form2 = form);
+            testDriver(form1, form2);
```

Comment:
`RunFormTest` closes and disposes the FormCommit instance.
So you rather should copy `SelectedDiff.Size` instead of `form`.

Reply:
I changed the tests as suggested (4 tests are affected by this). But the test results for `MessageEdit_remembers_geometry` and `SelectedDiff_remembers_geometry` are still the same. Honestly, I'd be surprised, if this had changed the results.

Reply:
Yes, you are right. It [fails](https://ci.appveyor.com/project/gitextensions/gitextensions/builds/30840767) even with `JoinPendingOperationsAsync` before and after.
I have no clue, what's different on AppVeyor.
Though you can try it on AppVeyor via RDP (ip & port in the AppVeyor console, pw per direct mail).

Reply:
Success! I could isolate and fix the problem. The difference between my machine and the CI system was that I changed the size of the commit dialog and this never happens on the CI system.

If the dialog size is never changed, the following expression evaluates always to true and the code that I changed originally is never called:

```csharp
if (SplitterSize == prevSize && Dpi == prevDpi)
{
    // Needed to be fixed for tests on AppVeyor 
    ...
}
else
{
    // Originally only changed here
    ...
}
```

The only tests that failed at the last build are ones that I didn't touched and they seem to have some kind of async problems.

Reply:
> The only tests that failed at the last build are ones that I didn't touched and they seem to have some kind of async problems.

We are working on a solution.

Reply:
> This behavior can be reproduced on the local machine, if you...

My sincere apologies, I started typing a very similar suggestion on a weekend, but I must have been distracted before I submitted it. Could have saved you few hours...

Reply:
@RussKie Ok. But at least I learnt some things about your CI system :)